### PR TITLE
Release 0.5.0

### PR DESCRIPTION
**Pull request by gouthamve**
I've gone through the history and the changelog is updated. One thing missing from it is https://github.com/cortexproject/cortex/pull/1750 but I've decided to explicitly exclude it until we have the WAL pieces and a guide to use the WAL documented.

It has been too long since 0.4.0 and I think it's time we get another release out. I know we thought of doing 1.0 but I don't have enough time to work on it yet and it's best done as the next release.

### CODE REVIEW

**Comment by codesome** [comment]
(Just a note that date needs to be changed in the CHANGELOG and RELEASE files before releasing)

**Comment by bboreham** [comment]
I suggest holding this release until someone can confirm that a 0.4-0.5 update with configs database works for them, unlike what I got at #1972

**Comment by bboreham** [comment]
Needs to shout louder about the Ruler changes - I had to roll back because we had a single instance doing all the work and now we need to scale out.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ## master / unreleased
 
+## 0.5.0 / 2020-01-15
+
+Note that the ruler flags need to be changed in this upgrade. You're moving from a single node ruler to something that can be sharded.
```
Comment (bboreham) [inline CHANGELOG.md]: This is good, but I suggest to call attention to the fact that you might _need_ to shard.  E.g. if you've set `ruler.num-workers` higher.

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,11 +29,23 @@
 * [ENHANCEMENT] Added `password` and `enable_tls` options to redis cache configuration. Enables usage of Microsoft Azure Cache for Redis service.
 * [BUGFIX] Fixed unnecessary CAS operations done by the HA tracker when the jitter is enabled. #1861
 * [BUGFIX] Fixed #1904 ingesters getting stuck in a LEAVING state after coming up from an ungraceful exit. #1921
+* [BUGFIX] Reduce memory usage when ingester Push() errors. #1922
 * [BUGFIX] TSDB: Fixed handling of out of order/bound samples in ingesters with the experimental TSDB blocks storage. #1864
 * [BUGFIX] TSDB: Fixed querying ingesters in `LEAVING` state with the experimental TSDB blocks storage. #1854
 * [BUGFIX] TSDB: Fixed error handling in the series to chunks conversion with the experimental TSDB blocks storage. #1837
 * [BUGFIX] TSDB: Fixed TSDB creation conflict with blocks transfer in a `JOINING` ingester with the experimental TSDB blocks storage. #1818
 
+### Upgrading Postgres
```
Comment (bboreham) [inline CHANGELOG.md]: Maybe clarify this is specific to the configs service?

File: VERSION [hunk 3]
```diff
@@ -1 +1 @@
-0.4.0
+0.5.0-rc.0
```
Comment (bboreham) [inline VERSION]: Should this be 0.5.0 for the release?

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,13 @@
 
 ## master / unreleased
 
+## 0.5.0-rc1 / 2020-01-15
+
+Note that the ruler flags need to be changed in this upgrade. You're moving from a single node ruler to something that might need to be sharded.
+If you are running with a high `-ruler.num-workers` and if you're not able to execute all your rules in 15s, then you'll need to shard.
```
Comment (pracucci) [inline CHANGELOG.md]: Where does this `15s` come from?

Comment (bboreham) [reply CHANGELOG.md]: It's the default for `-ruler.evaluation-interval`; I hadn't noticed it was configurable.

File: CHANGELOG.md [hunk 5]
```diff
@@ -22,11 +30,23 @@
...
 * [BUGFIX] TSDB: Fixed handling of out of order/bound samples in ingesters with the experimental TSDB blocks storage. #1864
 * [BUGFIX] TSDB: Fixed querying ingesters in `LEAVING` state with the experimental TSDB blocks storage. #1854
 * [BUGFIX] TSDB: Fixed error handling in the series to chunks conversion with the experimental TSDB blocks storage. #1837
 * [BUGFIX] TSDB: Fixed TSDB creation conflict with blocks transfer in a `JOINING` ingester with the experimental TSDB blocks storage. #1818
 
+### Upgrading Postgres
+
+Reference: https://github.com/golang-migrate/migrate/tree/master/database/postgres#upgrading-from-v1
+
+1. Install the migrate package cli tool: https://github.com/golang-migrate/migrate/tree/master/cmd/migrate#installation
```
Comment (bboreham) [inline CHANGELOG.md]: Looks like before this step we need to `DROP TABLE schema_migrations`.

File: CHANGELOG.md (hunk 4 above)
Comment (pracucci) [reply CHANGELOG.md]: Then it could be more clear if we mention it here.

**Comment by gouthamve** [comment]
@bboreham @pracucci @tomwilkie @csmarchbanks @khaines 

Can we merge this in? I've merged master into it and made sure we didn't miss any PRs or changes. I think we can merge this into master and I'll send an email to the mailing list that we're abandoning the 0.5 release but will kick off 0.6 next week.

**Comment by pstibrany** [comment]
> Can we merge this in? I've merged master into it and made sure we didn't miss any PRs or changes.

Why did you merge master to this branch? I think that it just creates weird log history, and merging release branch to master would be enough.

**Comment by gouthamve** [comment]
There were conflicts, I didn't want to resolve them via the UI and there
was no way to push to master from my local. I didn't see another way other
than the merge.

**Comment by pstibrany** [comment]
> and there was no way to push to master from my local

ok, I see. Perhaps rebasing would be another option, but that’s very atypical modification for  release branch.

**Comment by gouthamve** [comment]
Yeah don't want to force push to the release branch, some of the commits have been tagged.

File: RELEASE.md [hunk 6]
```diff
@@ -12,7 +12,8 @@ Our goal is to provide a new minor release every 4 weeks. This is a new process
 | v0.2.0         | 2019-08-28                                 | Goutham Veeramachaneni (Github: @gouthamve) |
 | v0.3.0         | 2019-10-09                                 | Bryan Boreham (@bboreham)                   |
 | v0.4.0         | 2019-11-13                                 | Tom Wilkie (@tomwilkie)                     |
-| v0.5.0         | 2019-12-11                                 | **searching for volunteer**                 |
+| v0.5.0         | 2020-01-08                                 | Goutham Veeramachaneni (@gouthamve)         |
```
Comment (pracucci) [inline RELEASE.md]: We should update this and mention it has been abandoned.

**Pull request merged**